Investigated and removed unused DLLs from the MVC_Sample and SPA_Sample projects

SandeepNayak077 opened the pull request:
Unused DLL in the MVC_Sample:
- log4net
- Ionic.zip
- Newtonsoft.Json

Unused DLL in the SPA_Sample
- log4net
- Ionic.zip

yusukemaegawa commented on root_VS2015/programs/C#/Samples/WebApp_sample/SPA_Sample/SPA_Sample/Web.config:
```diff
@@ -128,6 +128,10 @@
         <bindingRedirect oldVersion="0.0.0.0-3.0.0.0" newVersion="3.0.0.0" />
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="log4net" publicKeyToken="669e0ddf0bb1aa2a" culture="neutral" />
```
@SandeepNayak077 

Though log4net.dll is deleted, why is this setting necessary?

SandeepNayak077 commented:
@yusukemaegawa 

> https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82/files#r53128360

To avoid the following waring, that setting is necessary.

<b>Warning information:</b>
```
No way to resolve conflict between "log4net, Version=1.2.15.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a" and "log4net, Version=1.2.11.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a". Choosing "log4net, Version=1.2.15.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a" arbitrarily.
1>  Consider app.config remapping of assembly "log4net, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a" from Version "1.2.11.0" [] to Version "1.2.15.0" [C:\OpenTouryoTemplates\root_VS2015\programs\C#\Frameworks\Infrastructure\Build\log4net.dll] to solve conflict and get rid of warning.
```

Please let us know if there is any problem.

SandeepNayak077 commented:
@yusukemaegawa 
@daisukenishino 

Sorry, while commenting by mistake I clicked <b>Close and comment</b>.
I reopened this pull request.

Please let us know about [this comment](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82/#issuecomment-185138179).

yusukemaegawa commented:
@SandeepNayak077 

Thank you for your reply.
But, the following phenomenon does not reproduce in my environment.
(No warnings are shown even if the setting for log4net does not exist.)

><b>Warning information:</b>
>```
>No way to resolve conflict between "log4net, Version=1.2.15.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a" and "log4net, Version=1.2.11.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a". Choosing "log4net, Version=1.2.15.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a" arbitrarily.
>1>  Consider app.config remapping of assembly "log4net, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a" from Version "1.2.11.0" [] to Version "1.2.15.0" [C:\OpenTouryoTemplates\root_VS2015\programs\C#\Frameworks\Infrastructure\Build\log4net.dll] to solve conflict and get rid of warning.
>```

SandeepNayak077 commented:
@yusukemaegawa 

Oh is it.!
I wonder why that warning is appearing in our environment.
I tried to figure out how the solution is refering this version "1.2.11.0",
I did not get any information of this version so I adopted that countermeasures.

Do you want me to remove that settings?
please let us know.

yusukemaegawa commented:
@SandeepNayak077 

Oh dear...
I confirmed this pull request by the following procedure.

1. Downloaded zip file from [github.com](https://github.com/SymphonyTeleca/OpenTouryoTemplates/tree/nugetpkg_samp).
1. Unzipped the zip file to some folder.
1. Executed the *.bat file in root_VS2015.
1. Started Visual Studio 2015, and opened SPA_Sample.sln.
1. Removed the setting for log4net from web.config.
1. Rebuilt the SPA_Sample.sln.

---> No warnings are shown.

Will you try the above procedure?

SandeepNayak077 commented:
@yusukemaegawa 

Sure, I will try and report you.

SandeepNayak077 commented:
@yusukemaegawa 

> https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82#issuecomment-185568418

According to your steps I am not getting any warnings.

When I was testing, I merged the follwing two pull requests in my test branch
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/90
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82

Can you please test the sample by merging above two pull requests?
Because the above pull requests are necessory in the main branch.

If you still did not get any warnings, 
Then I will remove that code in the next pull request.

yusukemaegawa commented:
@SandeepNayak077 

I merged the following to pull requests manually, and this time's phenomenon reproduced. (The warning was shown.)
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/90
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82

[DotNetOpenAuth.Core](https://www.nuget.org/packages/DotNetOpenAuth.Core/) seems to use log4net internally.
Therefore, the confliction might occur between "the version of log4net that DotNetOpenAuth.Core uses" and "the version of log4net that OpenTouryo uses".
If I removed DotNetOpenAuth.Core package and the dependency packages, the warning disappeared.

Please confirm.

@daisukenishino 

If SPA_Sample does not use DotNetOpenAuth, how about removing these packages?

daisukenishino commented:
@yusukemaegawa 
If DotNetOpenAuth is not required, I will remove the DotNetOpenAuth from SPA.

@SandeepNayak077 
PullRequest of 82 and 90 has been merged to nishino branch.
If DotNetOpenAuth is not required, please send the PullRequest to modify the nishino branch.

SandeepNayak077 commented:
@daisukenishino 

In the SPA_sample,
There is an assemble reference of <b>Microsoft.Web.WebPages.OAuth</b> [here](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Samples/WebApp_sample/SPA_Sample/SPA_Sample/App_Start/AuthConfig.cs#L5).
The <b>Microsoft.Web.WebPages.OAuth</b> internally uses the <b>DotNetOpenAuth </b>.
If [this reference] (https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Samples/WebApp_sample/SPA_Sample/SPA_Sample/App_Start/AuthConfig.cs#L5) is required, then the <b>DotNetOpenAuth </b> is required.

<b>Work around:</b>
We deleted [this reference](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Samples/WebApp_sample/SPA_Sample/SPA_Sample/App_Start/AuthConfig.cs#L5) and removed the following packages.
- DotNetOpenAuth.AspNet
- DotNetOpenAuth.Core
- DotNetOpenAuth.OAuth.Consumer
- DotNetOpenAuth.OAuth.Core
- DotNetOpenAuth.OpenId.Core
- DotNetOpenAuth.OpenId.RelyingParty
- Microsoft.AspNet.WebPages.OAuth
- Microsoft.AspNet.WebPages.OAuth.ja

Build was successful with no errors or any kind of warnings. 

Is [this reference](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Samples/WebApp_sample/SPA_Sample/SPA_Sample/App_Start/AuthConfig.cs#L5) required in the SPA sample?
or
Shall we do the above work around and do code commit?
please confirm.

daisukenishino commented:
@SandeepNayak077 

I do not required this reference.
```CS
using Microsoft.Web.WebPages.OAuth;
```
Please send pull request after do the above workaround.
